### PR TITLE
[BEN-180] Rename resource template in chef-12 and update omnibus

### PR DESCRIPTION
**Pull request by jaymalasinha**
Signed-off-by: Jaymala Sinha <jsinha@chef.io>

### Description

This is a backport of #6286

Since the symlinks created inside /usr/bin conflict between our
"Angry" and mainstream projects on our testers, we are updating IPS
the packaging to override "projectname-symlinks.erb" if it exists.
We are leaving symlinks.erb for backward compatibilty. PRs with resource
filename updates in chef and omnibus-toolchain will follow this PR.

### Issues Resolved

Related omnibus PR: chef/omnibus#793

### Check List

- [x] New functionality includes tests
- [ ] All tests pass
- [ ] RELEASE\_NOTES.md, has been updated if required (not required for bugfixes, required for API changes)
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

@chef/engineering-services

### CODE REVIEW

File: omnibus/Gemfile.lock [hunk 1]
```diff
@@ -250,7 +243,6 @@ GEM
 
 PLATFORMS
   ruby
-  x86-mingw32
```
Comment (rhass) [inline omnibus/Gemfile.lock]: `-  x86-mingw32`  The windows stuff seemed to get chopped off of the Gemfile.lock -- I am not sure why, but this seems like a mistake.

**Comment by thommay** [comment]
Ah, it appears that this is a backport of #6286, which wasn't reviewed by any chef maintainers. Please see the maintenance policy for chef, and request review from @chef/client-maintainers before merging.

**Comment by coderanger** [comment]
Given the change, I'm kind of assuming that the goal was to allow the angrychef builds to use different symlink paths. That would not need to be backported as angrychef is presumably using 13.x by now?

**Comment by scotthain** [comment]
It is a backport of 6286 - which we push through quickly as it was blocking work. We'll be sure to notify/request review appropriately next time. As it stands right now, parts of our internal infrastructure are not on chef/angrychef 13 for a number of reasons, and we do need this in 12.

**Comment by jaymalasinha** [comment]
@rhass @scotthain @thommay @coderanger   I updated platform x86-mingw32 which got removed during omnibus update.

**Pull request merged**